app/src/main/java/com/duckduckgo/app/browser/trafficquality/remote/AndroidFeaturesHeaderProvider.kt

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.trafficquality.remote
+
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.autoconsent.api.Autoconsent
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.mobile.android.app.tracking.AppTrackingProtection
+import com.duckduckgo.networkprotection.api.NetworkProtectionState
+import com.duckduckgo.privacy.config.api.Gpc
+import com.squareup.anvil.annotations.ContributesBinding
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+import javax.inject.Inject
+import kotlinx.coroutines.runBlocking
+
+interface AndroidFeaturesHeaderProvider {
+    fun provide(): String?
+}
+
+@ContributesBinding(AppScope::class)
+class RealAndroidFeaturesHeaderProvider @Inject constructor(
+    private val appBuildConfig: AppBuildConfig,
+    private val featuresRequestHeaderStore: FeaturesRequestHeaderStore,
+    private val autoconsent: Autoconsent,
+    private val gpc: Gpc,
+    private val appTrackingProtection: AppTrackingProtection,
+    private val networkProtectionState: NetworkProtectionState,
+) : AndroidFeaturesHeaderProvider {
+    override fun provide(): String? {
+        val config = featuresRequestHeaderStore.getConfig()
+        if (config.isNotEmpty()) {
+            val versionConfig = config.find { it.appVersion == appBuildConfig.versionCode }
+            if (versionConfig != null) {
+                if (shouldLogValue(versionConfig)) {
+                    return mapFeatures(versionConfig)
+                }
+            } else {
+                return null
+            }
+        } else {
+            return null
+        }
+
+        return null
```

Comment:
This can be simplified. The `isNotEmpty` check isn't needed, `find` will return `null` if it’s empty:

```suggestion
        val versionConfig = config.find { it.appVersion == appBuildConfig.versionCode }
        return if (versionConfig != null && shouldLogValue(versionConfig)) {
            mapFeatures(versionConfig)
        } else {
            null
        }
```